src/main/java/reposense/parser/PathLocationParser.java

```diff
@@ -0,0 +1,72 @@
+package reposense.parser;
+
+import static reposense.util.FileUtil.fileExists;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+/**
+ * Contains functionality to parse repo location details from a path to a repo.
+ */
+public class PathLocationParser {
+
+    private static final String GIT_LINK_SUFFIX = ".git";
+    private static final String BRANCH_DELIMITER = "#";
+    /**
+     * Parses a given path to a repo and returns an array containing the following info:
+     * { path to repo, repository name, organisation name, branch name (if any) }
+     *
+     * @param pathAndBranch A path to a repository, with a branch name optionally appended at the end
+     *         e.g. /home/users/tom/Desktop/reposense#release
+     * @param isPathValidationNeeded If this flag is set to true, then the method will check
+     *         verify that {@code path} is a valid path and refers to an actual directory on the
+     *         filesystem.
+     *
+     * @return null if the given String is an invalid path, or no directory exists at the path,
+     *         and {@code isPathValidationNeeded} was set to true.
+     */
+    public static String[] tryParsingAsPath(String pathAndBranch, boolean isPathValidationNeeded)  {
```

Comment:
No need to specify `try` in the method name, `parseAsPath` is clear enough.  A normal workflow should be:
```
try:
  `parseAsPath`
```

Reply:
Sure, I've changed the name. Just to be clear, the `parseAsPath` method doesn't throw an exception.